amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseRefresher.java

```diff
@@ -292,7 +292,13 @@ long sleep(long timeToSleepMillis) {
     @Override
     public boolean isLeaseTableEmpty()
             throws DependencyException, InvalidStateException, ProvisionedThroughputException {
-        return list(1, 1, null).isEmpty();
+        return isLeaseTableEmptyForStreamIdentifier(null);
+    }
+
+    @Override
+    public boolean isLeaseTableEmptyForStreamIdentifier(StreamIdentifier streamIdentifier)
+            throws DependencyException, ProvisionedThroughputException, InvalidStateException {
+        return list(1, 1, streamIdentifier).isEmpty();
```

Comment:
As called out earlier, can you check if this would return false even if we have a streamIdentifier entry as a last record in a table with say million records?